libraries/textcomposer/impl/src/main/kotlin/io/element/android/libraries/textcomposer/TextComposerLinkDialog.kt

```diff
@@ -0,0 +1,227 @@
+    onCreateLinkRequest: (url: String, text: String) -> Unit,
+    modifier: Modifier = Modifier,
+) {
+    var linkText by remember { mutableStateOf("") }
+    var linkUrl by remember { mutableStateOf("") }
+
+    val titleText = stringResource(CommonStrings.rich_text_editor_create_link)
+
+    fun onSubmit() {
+        onCreateLinkRequest(linkUrl, linkText)
+        onDismissRequest()
+    }
+
+    ListDialog(
+        onDismissRequest = onDismissRequest,
+        onSubmit = ::onSubmit,
+        title = titleText,
+        modifier = modifier
+    ) {
+        item {
+            TextFieldListItem(
+                placeholder = "Text",
+                text = linkText,
+                onTextChanged = { linkText = it },
+            )
+        }
+        item {
+            TextFieldListItem(
+                placeholder = "Link",
+                text = linkUrl,
+                onTextChanged = { linkUrl = it },
+            )
+        }
+    }
+}
+
+@Composable
+private fun CreateLinkWithoutTextDialog(
+    onDismissRequest: () -> Unit,
+    onSaveLinkRequest: (url: String) -> Unit,
+    modifier: Modifier = Modifier,
+) {
+    var linkUrl by remember { mutableStateOf("") }
+
+    val titleText = stringResource(CommonStrings.rich_text_editor_create_link)
+
+    fun onSubmit() {
+        onSaveLinkRequest(linkUrl)
+        onDismissRequest()
+    }
+
+    ListDialog(
+        onDismissRequest = onDismissRequest,
+        onSubmit = ::onSubmit,
+        title = titleText,
+        modifier = modifier
+    ) {
+        item {
+            TextFieldListItem(
+                placeholder = "Link",
+                text = linkUrl,
+                onTextChanged = { linkUrl = it },
+            )
+        }
+    }
+}
+
+// The edit link dialog does not yet support displaying or editing the text of a link
+// https://github.com/matrix-org/matrix-rich-text-editor/issues/617
+@Composable
+private fun EditLinkDialog(
+    currentUrl: String,
+    onDismissRequest: () -> Unit,
+    onSaveLinkRequest: (url: String) -> Unit,
+    onRemoveLinkRequest: () -> Unit,
+    modifier: Modifier = Modifier,
+) {
+    var linkUrl by remember { mutableStateOf(currentUrl) }
+
+    val titleText = stringResource(CommonStrings.rich_text_editor_edit_link)
+
+    fun onSubmit() {
+        onSaveLinkRequest(linkUrl)
+        onDismissRequest()
+    }
+
+    fun onRemoveClicked() {
+        onRemoveLinkRequest()
+        onDismissRequest()
+    }
+
+    ListDialog(
+        onDismissRequest = onDismissRequest,
+        onSubmit = ::onSubmit,
+        title = titleText,
+        modifier = modifier
+    ) {
+        item {
+            TextFieldListItem(
+                placeholder = "Link",
+                text = linkUrl,
+                onTextChanged = { linkUrl = it },
+            )
+        }
+        item {
+            ListItem(
+                headlineContent = { Text("Remove link", color = ElementTheme.colors.textCriticalPrimary) }, // TODO
```

Comment:
Maybe a bit more info about the TODO or move it to the comment for this composable?